src/makeInfinityAdder.js

```diff
@@ -6,7 +6,21 @@
  */
 
 function makeInfinityAdder() {
-  // write code here
+  const sum = (x) => {
+    if (typeof x !== 'undefined') {
+      return (y) => {
+        if (typeof y !== 'undefined') {
+          return sum(x + y);
+        } else {
+          return x;
+        }
+      };
+    }
+
+    return 0;
```

Comment:
I think it's better to swap this statements and return 0 if (x === undefined), else - do main logic

Reply:
makes sense to fail fast, thanks